core/src/main/java/org/apache/iceberg/actions/FileURI.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.net.URI;
+import java.util.Map;
+import org.apache.hadoop.fs.Path;
```

Comment:
Any chance we can leave Hadoop dependencies out of this class since it's essentially just acting as a container class? For example we could provide public FileUri(URI uri), then a caller can use new org.apache.hadoop.fs.Path(uriString).

Just don't want to spread Hadoop-classes around the project if possible